web3_proxy/src/rpcs/request.rs

```diff
@@ -299,87 +308,127 @@ impl OpenRequestHandle {
                 if let Some(msg) = msg {
                     if msg.starts_with("execution reverted") {
                         trace!("revert from {}", self.rpc);
-                        ResponseTypes::Revert
+                        ResponseErrorType::Revert
                     } else if msg.contains("limit") || msg.contains("request") {
                         trace!("rate limit from {}", self.rpc);
-                        ResponseTypes::RateLimit
+                        ResponseErrorType::RateLimit
                     } else {
-                        ResponseTypes::Ok
+                        ResponseErrorType::Error
                     }
                 } else {
-                    ResponseTypes::Ok
+                    ResponseErrorType::Error
                 }
             } else {
-                ResponseTypes::Ok
+                ResponseErrorType::Error
             };
 
-            if matches!(response_type, ResponseTypes::RateLimit) {
-                if let Some(hard_limit_until) = self.rpc.hard_limit_until.as_ref() {
-                    let retry_at = Instant::now() + Duration::from_secs(1);
+            match response_type {
+                ResponseErrorType::RateLimit => {
+                    if let Some(hard_limit_until) = self.rpc.hard_limit_until.as_ref() {
+                        // TODO: how long? different providers have different rate limiting periods, though most seem to be 1 second
+                        // TODO: until the next second, or wait 1 whole second?
+                        let retry_at = Instant::now() + Duration::from_secs(1);
 
-                    trace!("retry {} at: {:?}", self.rpc, retry_at);
+                        trace!("retry {} at: {:?}", self.rpc, retry_at);
```

Comment:
if you want to change the `matches` to this single match block, then this shouldn't always be trace. if `error_handler` is set to a higher level, we need to see that this is getting rate limited at the matching level. it was split into a `matches` and a `match` so that the log level code didn't need to be duplicated

Reply:
i'm fine with either, let me know what you prefer? this again (similar to the above one), occurs because stats_v2 and main simultaneously (and separate) modified this file. I will check if I can fix this

Reply:
This should be resolved now? Took over changes from main